core/src/layers/complete.rs

```diff
@@ -382,7 +389,7 @@ impl<A: Access> LayeredAccess for CompleteAccessor<A> {
 
     // Todo: May move the logic to the implement of Layer::layer of CompleteAccessor<A>
     fn metadata(&self) -> Arc<AccessorInfo> {
-        let mut meta = (*self.meta).clone();
+        let mut meta = self.meta.as_ref().clone();
         let cap = meta.full_capability_mut();
         if cap.list && cap.write_can_empty {
```

Comment:
Maybe we can remove those code and the comment?

Reply:
if we do this, the issue #4888 is not resolved....

Reply:
> if we do this, the issue #4888 is not resolved....

Could you elaborate further? I'm not sure which issue remains unresolved.

Reply:
> Maybe we can remove those code and the comment?

you means delete the 
```
// Todo: May move the logic to the implement of Layer::layer of CompleteAccessor<A>
```
and 
```
let mut meta = self.meta.as_ref().clone();
```
?

Reply:
Yep, since we have created the `Arc<AccessInfo>` during `layer()`, we can remove the dup code here.